File: app/src/main/java/com/eazy/stcbusiness/utils/listener/CustomSetOnClickViewListener.java
package com.eazy.stcbusiness.utils.listener;

import android.view.View;


public class CustomSetOnClickViewListener implements View.OnClickListener {
    private final CustomResponseOnClickListener customResponseOnClickListener;

    public CustomSetOnClickViewListener(CustomResponseOnClickListener customResponseOnClickListener){
        this.customResponseOnClickListener = customResponseOnClickListener;
    }

    @Override
    public void onClick(View view) {
        view.setEnabled(false);
        view.postDelayed(() -> view.setEnabled(true),500);
        customResponseOnClickListener.onClick(view);
    }
}

File: app/src/main/java/com/eazy/stcbusiness/utils/listener/CustomSetOnClickViewListener.kt
package com.eazy.stcbusiness.utils.listener

import android.view.View

class CustomSetOnClickViewListener
    (private val customResponseOnClickListener: CustomResponseOnClickListener) : View.OnClickListener {

    override fun onClick(view: View) {
        view.isEnabled = false
        view.postDelayed({ view.isEnabled = true }, 500)
        customResponseOnClickListener.onClick(view)
    }
}